gateway-js/src/__tests__/loadSupergraphSdlFromStorage.test.ts

```diff
@@ -23,402 +30,69 @@ describe('loadSupergraphSdlFromStorage', () => {
     const result = await loadSupergraphSdlFromStorage({
       graphRef,
       apiKey,
-      endpoint: mockCloudConfigUrl,
+      endpoint: mockCloudConfigUrl1,
+      errorReportingEndpoint: mockOutOfBandReporterUrl,
       fetcher,
       compositionId: null,
+    });
+    expect(result).toMatchInlineSnapshot(sampleSchema);
+  });
 
+  it('Queries alternate Uplink URL if first one fails', async () => {
+    mockSupergraphSdlRequest('originalId-1234').reply(500);
+    mockOutOfBandReportRequestSuccess();
+    mockSupergraphSdlRequestIfAfter('originalId-1234', mockCloudConfigUrl2).reply(
+      200,
+      JSON.stringify({
+        data: {
+          routerConfig: {
+            __typename: 'RouterConfigResult',
+            id: 'originalId-1234',
+            supergraphSdl: getTestingSupergraphSdl()
+          },
+        },
+      }),
+    );
+
+    const fetcher = getDefaultFetcher();
+    const result = await loadSupergraphSdlFromUplinks({
+      graphRef,
+      apiKey,
+      endpoints: [mockCloudConfigUrl1, mockCloudConfigUrl2],
+      errorReportingEndpoint: mockOutOfBandReporterUrl,
```

Comment:
I just realized if we leave this option out (maybe set to `undefined`?) then we shouldn't have to mock the reporter calls as well. I think it's a separate concern from this test (though maybe we should have a test which asserts that a report is sent when failures happen if it doesn't exist)

Reply:
I set `errorReportingEndpoint` to undefined here.